src/main/java/com/example/Customer.java

```diff
@@ -0,0 +1,30 @@
+package com.example;
+
+/**
+ * A customer is a person who visits a shop for buying goods.
+ *
+ * <p>The customer can have a loyalty card that has a specific number.
+ */
+public final class Customer {
+    /**
+     * In order to use previously collected bonuses, the customer uses the card during a purchase.
+     *
+     * <p>The absence of a card does not prevent the customer from purchasing the goods.
```

Comment:
Please move this documentation into the class header, and have a shorter comment on the card.